packages/api/src/decorator/private/templateMiddleware.ts

```diff
@@ -0,0 +1,47 @@
+import { warnOnce } from 'botframework-webchat-core';
+import { createChainOfResponsibility, type ComponentMiddleware } from 'react-chain-of-responsibility';
+import { type EmptyObject } from 'type-fest';
+import { any, array, custom, safeParse, type Output } from 'valibot';
+
+export default function createMiddlewareFacility<Props extends {} = EmptyObject, Request extends {} = EmptyObject>(
+  name: string
+) {
+  type Middleware = ComponentMiddleware<Request, Props>;
+
+  const validateMiddleware = custom<Middleware>(input => typeof input === 'function', 'Middleware must be a function.');
+
+  const middlewareSchema = array(any([validateMiddleware]));
+
+  const isMiddleware = (middleware: unknown): middleware is Output<typeof middlewareSchema> =>
+    safeParse(middlewareSchema, middleware).success;
+
+  const warnInvalid = warnOnce(`"${name}" prop is invalid`);
+
+  const rectifyProps = (middleware: unknown): readonly Middleware[] => {
```

Comment:
The rectification now is... "freeze the middleware prop if it is not frozen".

Do we want to expose the rectification and ask the consumer of `templateMiddleware` to rectify on their own?

Or should we auto-rectify with the exported Provider?

Reply:
We should combine this into init and add to `templateMiddleware`